File: implementation/scipamato/scipamato-core-persistence-jooq/src/test/java/ch/difty/scipamato/persistence/search/SearchOrderUpdateSetStepSetterTest.java
package ch.difty.scipamato.persistence.search;

import static ch.difty.scipamato.db.tables.SearchOrder.*;
import static ch.difty.scipamato.persistence.search.SearchOrderRecordMapperTest.*;
import static org.mockito.Mockito.*;

import org.mockito.Mock;

import ch.difty.scipamato.db.tables.records.SearchOrderRecord;
import ch.difty.scipamato.entity.SearchOrder;
import ch.difty.scipamato.persistence.UpdateSetStepSetter;
import ch.difty.scipamato.persistence.UpdateSetStepSetterTest;

public class SearchOrderUpdateSetStepSetterTest extends UpdateSetStepSetterTest<SearchOrderRecord, SearchOrder> {

    private final UpdateSetStepSetter<SearchOrderRecord, SearchOrder> setter = new SearchOrderUpdateSetStepSetter();

    @Mock
    private SearchOrder entityMock;

    @Override
    protected UpdateSetStepSetter<SearchOrderRecord, SearchOrder> getSetter() {
        return setter;
    }

    @Override
    protected SearchOrder getEntity() {
        return entityMock;
    }

    @Override
    protected void specificTearDown() {
        verifyNoMoreInteractions(entityMock);
    }

    @Override
    protected void entityFixture() {
        when(entityMock.getId()).thenReturn(ID);
        SearchOrderRecordMapperTest.entityFixtureWithoutIdFields(entityMock);
    }

    @Override
    protected void stepSetFixtureExceptAudit() {
        when(getStep().set(SEARCH_ORDER.NAME, NAME)).thenReturn(getMoreStep());
        when(getMoreStep().set(SEARCH_ORDER.OWNER, OWNER)).thenReturn(getMoreStep());
        when(getMoreStep().set(SEARCH_ORDER.GLOBAL, GLOBAL)).thenReturn(getMoreStep());
    }

    @Override
    protected void stepSetFixtureAudit() {
        when(getMoreStep().set(SEARCH_ORDER.CREATED, CREATED)).thenReturn(getMoreStep());
        when(getMoreStep().set(SEARCH_ORDER.CREATED_BY, CREATED_BY)).thenReturn(getMoreStep());
        when(getMoreStep().set(SEARCH_ORDER.LAST_MODIFIED, LAST_MOD)).thenReturn(getMoreStep());
        when(getMoreStep().set(SEARCH_ORDER.LAST_MODIFIED_BY, LAST_MOD_BY)).thenReturn(getMoreStep());
        when(getMoreStep().set(SEARCH_ORDER.VERSION, VERSION + 1)).thenReturn(getMoreStep());
    }

    @Override
    protected void verifyCallToAllFieldsExceptAudit() {
        verify(entityMock).getName();
        verify(entityMock).getOwner();
        verify(entityMock).isGlobal();
    }

    @Override
    protected void verifyStepSettingExceptAudit() {
        verify(getStep()).set(SEARCH_ORDER.NAME, NAME);
        verify(getMoreStep()).set(SEARCH_ORDER.OWNER, OWNER);
        verify(getMoreStep()).set(SEARCH_ORDER.GLOBAL, GLOBAL);
    }

    @Override
    protected void verifyStepSettingAudit() {
        verify(getMoreStep()).set(SEARCH_ORDER.CREATED, CREATED);
        verify(getMoreStep()).set(SEARCH_ORDER.CREATED_BY, CREATED_BY);
        verify(getMoreStep()).set(SEARCH_ORDER.LAST_MODIFIED, LAST_MOD);
        verify(getMoreStep()).set(SEARCH_ORDER.LAST_MODIFIED_BY, LAST_MOD_BY);
        verify(getMoreStep()).set(SEARCH_ORDER.VERSION, VERSION + 1);
    }

}

File: core/core-persistence-jooq/src/test/kotlin/ch/difty/scipamato/core/persistence/search/SearchOrderUpdateSetStepSetterTest.kt
package ch.difty.scipamato.core.persistence.search

import ch.difty.scipamato.core.db.tables.SearchOrder.SEARCH_ORDER
import ch.difty.scipamato.core.db.tables.records.SearchOrderRecord
import ch.difty.scipamato.core.entity.search.SearchOrder
import ch.difty.scipamato.core.persistence.RecordMapperTest
import ch.difty.scipamato.core.persistence.UpdateSetStepSetter
import ch.difty.scipamato.core.persistence.UpdateSetStepSetterTest
import io.mockk.confirmVerified
import io.mockk.every
import io.mockk.mockk
import io.mockk.verify

internal class SearchOrderUpdateSetStepSetterTest : UpdateSetStepSetterTest<SearchOrderRecord, SearchOrder>() {

    override val setter: UpdateSetStepSetter<SearchOrderRecord, SearchOrder> = SearchOrderUpdateSetStepSetter()

    override val entity = mockk<SearchOrder>()

    override fun specificTearDown() {
        confirmVerified(entity)
    }

    override fun entityFixture() {
        SearchOrderRecordMapperTest.entityFixtureWithoutIdFields(entity)
    }

    override fun stepSetFixtureExceptAudit() {
        every { step.set(SEARCH_ORDER.NAME, SearchOrderRecordMapperTest.NAME) } returns moreStep
        every { moreStep.set(SEARCH_ORDER.OWNER, SearchOrderRecordMapperTest.OWNER) } returns moreStep
        every { moreStep.set(SEARCH_ORDER.GLOBAL, SearchOrderRecordMapperTest.GLOBAL) } returns moreStep
    }

    override fun stepSetFixtureAudit() {
        every { moreStep.set(SEARCH_ORDER.CREATED, RecordMapperTest.CREATED) } returns moreStep
        every { moreStep.set(SEARCH_ORDER.CREATED_BY, RecordMapperTest.CREATED_BY) } returns moreStep
        every { moreStep.set(SEARCH_ORDER.LAST_MODIFIED, RecordMapperTest.LAST_MOD) } returns moreStep
        every { moreStep.set(SEARCH_ORDER.LAST_MODIFIED_BY, RecordMapperTest.LAST_MOD_BY) } returns moreStep
        every { moreStep.set(SEARCH_ORDER.VERSION, RecordMapperTest.VERSION + 1) } returns moreStep
    }

    override fun verifyCallToAllFieldsExceptAudit() {
        verify { entity.name }
        verify { entity.owner }
        verify { entity.isGlobal }
    }

    override fun verifyStepSettingExceptAudit() {
        verify { step.set(SEARCH_ORDER.NAME, SearchOrderRecordMapperTest.NAME) }
        verify { moreStep.set(SEARCH_ORDER.OWNER, SearchOrderRecordMapperTest.OWNER) }
        verify { moreStep.set(SEARCH_ORDER.GLOBAL, SearchOrderRecordMapperTest.GLOBAL) }
    }

    override fun verifyStepSettingAudit() {
        verify { moreStep.set(SEARCH_ORDER.CREATED, RecordMapperTest.CREATED) }
        verify { moreStep.set(SEARCH_ORDER.CREATED_BY, RecordMapperTest.CREATED_BY) }
        verify { moreStep.set(SEARCH_ORDER.LAST_MODIFIED, RecordMapperTest.LAST_MOD) }
        verify { moreStep.set(SEARCH_ORDER.LAST_MODIFIED_BY, RecordMapperTest.LAST_MOD_BY) }
        verify { moreStep.set(SEARCH_ORDER.VERSION, RecordMapperTest.VERSION + 1) }
    }
}
